lib/gui/app/components/source-selector/source-selector.tsx

```diff
@@ -73,36 +101,145 @@ function getState() {
 	};
 }
 
-interface ImageSelectorProps {
+const URLSelector = ({ done }: { done: (imageURL: string) => void }) => {
+	const [imageURL, setImageURL] = React.useState('');
+	const [recentImages, setRecentImages]: [
+		string[],
+		(value: React.SetStateAction<string[]>) => void,
+	] = React.useState([]);
+	React.useEffect(() => {
+		const fetchRecentUrlImages = async () => {
+			const recentUrlImages: string[] = await getRecentUrlImages();
+			if (!Array.isArray(recentUrlImages)) {
+				setRecentImages([]);
```

Comment:
The array check and default empty list value should be done in `getRecentUrlImages`